django/core/management/utils.py

```diff
@@ -161,15 +162,20 @@ def find_formatters():
     return {"black_path": shutil.which("black")}
 
 
-def run_formatters(written_files, black_path=(sentinel := object())):
+def run_formatters(
+    written_files, black_path=(sentinel := object()), *, stderr=sys.stderr
+):
     """
     Run the black formatter on the specified files.
     """
     # Use a sentinel rather than None, as which() returns None when not found.
     if black_path is sentinel:
         black_path = shutil.which("black")
     if black_path:
-        subprocess.run(
-            [black_path, "--fast", "--", *written_files],
-            capture_output=True,
-        )
+        try:
+            subprocess.run(
+                [black_path, "--fast", "--", *written_files],
+                capture_output=True,
+            )
+        except OSError:
+            stderr.write(f"Black failed to launch.\n{sys.exc_info()}")
```

Comment:
Thank you for this change, I agree that using the command's stderr is a good and clever idea.
Could you please provide an example of how this looks? I just tried it and I think it needs some polishing on what is shown:
```
Black failed to launch.
(<class 'PermissionError'>, PermissionError(13, 'Permission denied'), <traceback object at 0x79aa4580b040>)
```

A few suggestions:
* Print "Formatters" instead of "Black" to follow the current naming pattern.
* Have a colon instead of the ending dot, when the traceback follows.
* Perhaps use `traceback.print_exc(file=stderr)` for printing the exception?

Reply:
Diff proposal:
```suggestion
            stderr.write(f"Formatters failed to launch:")
            traceback.print_exc(file=stderr)
```

Which outputs:
```
Migrations for 'testapp':
  testapp/migrations/0006_alter_basicmodel_name.py
    ~ Alter field name on basicmodel
Formatters failed to launch:
Traceback (most recent call last):

  File "/home/nessita/fellowship/django/django/core/management/utils.py", line 179, in run_formatters
    subprocess.run(
    ~~~~~~~~~~~~~~^
        [black_path, "--fast", "--", *written_files],
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        capture_output=True,
        ^^^^^^^^^^^^^^^^^^^^
    )
    ^

  File "/usr/lib/python3.13/subprocess.py", line 554, in run
    with Popen(*popenargs, **kwargs) as process:
         ~~~~~^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/lib/python3.13/subprocess.py", line 1036, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                        pass_fds, cwd, env,
                        ^^^^^^^^^^^^^^^^^^^
    ...<5 lines>...
                        gid, gids, uid, umask,
                        ^^^^^^^^^^^^^^^^^^^^^^
                        start_new_session, process_group)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/lib/python3.13/subprocess.py", line 1851, in _execute_child
    self._posix_spawn(args, executable, env, restore_signals, close_fds,
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                      p2cread, p2cwrite,
                      ^^^^^^^^^^^^^^^^^^
                      c2pread, c2pwrite,
                      ^^^^^^^^^^^^^^^^^^
                      errread, errwrite)
                      ^^^^^^^^^^^^^^^^^^

  File "/usr/lib/python3.13/subprocess.py", line 1795, in _posix_spawn
    self.pid = os.posix_spawn(executable, args, env, **kwargs)
               ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

PermissionError: [Errno 13] Permission denied: '/home/nessita/fellowship/django/tests/user_commands/test_files/black'
```